timdex_dataset_api/dataset.py

```diff
@@ -0,0 +1,118 @@
+"""timdex_dataset_api/dataset.py"""
+
+import time
+
+import boto3
+import pyarrow as pa
+import pyarrow.dataset as ds
+from pyarrow import fs
+
+from timdex_dataset_api.config import configure_logger
+from timdex_dataset_api.exceptions import DatasetNotLoadedError
+
+logger = configure_logger(__name__)
+
+TIMDEX_DATASET_SCHEMA = pa.schema(
+    (
+        pa.field("timdex_record_id", pa.string()),
+        pa.field("source_record", pa.binary()),
+        pa.field("transformed_record", pa.binary()),
+        pa.field("source", pa.string()),
+        pa.field("run_date", pa.date32()),
+        pa.field("run_type", pa.string()),
+        pa.field("run_id", pa.string()),
+        pa.field("action", pa.string()),
+    )
+)
```

Comment:
Theoretically, this becomes the canonical definition of what the TIMDEX parquet dataset in S3 should be. 
 Whenever we read or write to it, we'll use this schema when invoking pyarrow.

It might make sense to break this out into a standalone file like `schema.py`, but opted to keep it simple and colocated with the code that will use it for the time being.

Reply:
I do think it's importance would be highlighted better in `schema.py` but as you said, this is fine for now

Reply:
Yeah, still on the fence a bit.  I think given the file module is `dataset` it's kind of accurate that this lives there, but maybe time will tell.